src/components/BookList.spec.js

```diff
@@ -62,8 +69,34 @@ describe('Book list page', () => {
         });
 
         it('should show that the book is available', () => {
-            expect(rendered.queryByTitle('Available')).toBeVisible();
+            rendered.queryAllByTitle('Available').forEach((availableTag) => {
+                expect(availableTag).toBeVisible();
+            });
         });
+
+        it('should let a user filter for books by title', async () => {
```

Comment:
This it block is testing and doing a lot of things. It might make it hard for future devs to easily know what to modify when functionality needs to change or what went wrong if the test fails. This is typically why we see only one assertion per test.

One strategy I've found valuable when testing React applications is to have a `describe` block that explains the state of the system that is being tested. In this case something like "User types a query into the search box". Then in the beforeEach setup that situation. Then you can have a bunch of `it` blocks that are named based on what you are asserting it('should not show filtered results'). it('should show matching results') .etc

Thoughts?

Reply:
I really like this approach! Thank you for the suggestion. I'll use it in my update.